src/badges/views.py

```
@@ -67,6 +70,74 @@ def badge_list(request):
+        the count before granting. Only published badges can be auto-granted.
+        """
+        response = super().form_valid(form)
+        if self.object.published:
+            badge_name = SiteConfig.get().custom_name_for_badge.lower()
+            grant_url = reverse('badges:grant_qualifying', args=[self.object.id])
+            messages.info(
+                self.request,
+                mark_safe(
+                    f'Prerequisites changed. '
+                    f'<a href="{grant_url}">Check and grant this {badge_name} to qualifying students?</a>'
+                )
```

Ruff (0.15.21)

[error] Use of `mark_safe` may expose cross-site scripting vulnerabilities

(S308)